### PR TITLE
[CEDS-2947] Updating Validation Error Messages Additional work

### PR DESCRIPTION
**Pull request by glidester**
Update text and remove full stops from end of validation error
messages.

Add separate error messages for empty vs invalid values.

Refactoring of the MucrOptions model class.

### CODE REVIEW

File: app/forms/AssociateUcr.scala [hunk 1]
```diff
@@ -48,8 +48,20 @@ object AssociateUcr {
...
     Forms.mapping(
       "kind" -> of[UcrType](UcrType.formatter),
-      "ducr" -> mandatoryIfEqual("kind", Ducr.formValue, text().verifying("ducr.error.format", validDucrIgnoreCase)),
-      "mucr" -> mandatoryIfEqual("kind", Mucr.formValue, text().verifying("mucr.error.format", validMucrIgnoreCase))
+      "ducr" -> mandatoryIfEqual(
+        "kind",
+        Ducr.formValue,
+        text()
+          .verifying("associate.ucr.ducr.error.empty", nonEmpty)
+          .verifying("associate.ucr.ducr.error.invalid", isEmpty or validDucrIgnoreCase)
```
Comment (udinobi) [inline app/forms/AssociateUcr.scala:56]: Do you need to double check if empty?

File: app/forms/AssociateUcr.scala [hunk 2]
```diff
@@ -48,8 +48,20 @@ object AssociateUcr {
...
+      ),
+      "mucr" -> mandatoryIfEqual(
+        "kind",
+        Mucr.formValue,
+        text()
+          .verifying("associate.ucr.mucr.error.empty", nonEmpty)
+          .verifying("associate.ucr.mucr.error.invalid", isEmpty or validMucrIgnoreCase)
```
Comment (udinobi) [inline app/forms/AssociateUcr.scala:63]: Do you actually need to double check if empty?

File: app/forms/DucrPartDetails.scala [hunk 3]
```diff
@@ -51,8 +51,12 @@ object DucrPartDetails {
     def bind(ducr: String, ducrPartId: String): DucrPartDetails = DucrPartDetails(ducr.toUpperCase, ducrPartId.toUpperCase)
 
     Forms.mapping(
-      "ducr" -> text().verifying("ducrPartDetails.ducr.error", validDucrIgnoreCase),
-      "ducrPartId" -> text().verifying("ducrPartDetails.ducrPartId.error", isValidDucrPartId)
+      "ducr" -> text()
+        .verifying("ducrPartDetails.ducr.empty", nonEmpty)
+        .verifying("ducrPartDetails.ducr.invalid", isEmpty or validDucrIgnoreCase),
```
Comment (udinobi) [inline app/forms/DucrPartDetails.scala:56]: Do you actually need to double check if empty?

File: app/forms/DucrPartDetails.scala [hunk 4]
```diff
@@ -51,8 +51,12 @@ object DucrPartDetails {
...
+      "ducrPartId" -> text()
+        .verifying("ducrPartDetails.ducrPartId.empty", nonEmpty)
+        .verifying("ducrPartDetails.ducrPartId.invalid", isEmpty or isValidDucrPartId)
```
Comment (udinobi) [inline app/forms/DucrPartDetails.scala:59]: Do you actually need to double check if empty?

File: app/forms/MucrOptions.scala [hunk 5]
```diff
@@ -40,37 +37,42 @@ object MucrOptions {
...
-        form2Model
-      )(model2Form)
+      .mapping(
+        "createOrAdd" -> requiredRadio("mucrOptions.error.unselected"),
+        "newMucr" -> mandatoryIfEqual(
+          "createOrAdd",
+          Create,
+          text()
+            .verifying("mucrOptions.reference.value.error.empty", nonEmpty)
+            .verifying("mucrOptions.reference.value.error.invalid", isEmpty or validMucrIgnoreCase)
```
Comment (udinobi) [inline app/forms/MucrOptions.scala:65]: Do you actually need to double check if empty?

File: app/forms/MucrOptions.scala [hunk 6]
```diff
@@ -40,37 +37,42 @@ object MucrOptions {
...
+        ),
+        "existingMucr" -> mandatoryIfEqual(
+          "createOrAdd",
+          Add,
+          text()
+            .verifying("mucrOptions.reference.value.error.empty", nonEmpty)
+            .verifying("mucrOptions.reference.value.error.invalid", isEmpty or validMucrIgnoreCase)
```
Comment (udinobi) [inline app/forms/MucrOptions.scala:72]: Do you actually need to double check if empty?

File: app/forms/AssociateUcr.scala (hunk 1 above)
Comment (glidester) [reply app/forms/AssociateUcr.scala:56]: They are not checking empty twice, the first verifying checks for if empty, the second verifying checks value is either empty or a valid value. This just prevents empty values from triggering the display of both empty and invalid error messages at the same time.

**Review by udinobi — APPROVED** [review]

**Pull request merged**